tortoise/tortoise.go

```diff
@@ -457,11 +470,32 @@ func (t *turtle) verifyLayers() error {
 			break
 		}
 		verified = target
+		for _, block := range t.layers[target].blocks {
+			if !block.dirty {
+				continue
+			}
+			// record range of layers where opinion has changed.
+			// once those layers fall out of hdist window - opinion can be recomputed
+			if block.validity != block.hare {
```

Comment:
if hare changed opinion and memory state was lost it doesn't matter, since hash will be recomputed on startup using new opinion

there could be a bug if opinion wasn't according to hare and that state got lost, will add a test

Reply:
found couple of bugs, must be robust now